yt_dlp/networking/_curlcffi.py

```diff
@@ -97,45 +98,72 @@ def read(self, amt=None):
             return self.fp.read(amt)
         except curl_cffi.requests.errors.RequestsError as e:
             if e.code == CurlECode.PARTIAL_FILE:
-                content_length = int_or_none(e.response.headers.get('Content-Length'))
+                content_length = e.response and int_or_none(e.response.headers.get('Content-Length'))
                 raise IncompleteRead(
                     partial=self.fp.bytes_read,
                     expected=content_length - self.fp.bytes_read if content_length is not None else None,
                     cause=e) from e
             raise TransportError(cause=e) from e
 
 
+# See: https://github.com/lexiforest/curl_cffi?tab=readme-ov-file#supported-impersonate-browsers
+#      https://github.com/lexiforest/curl-impersonate?tab=readme-ov-file#supported-browsers
+BROWSER_TARGETS: dict[tuple[int, ...], dict[str, ImpersonateTarget]] = {
+    (0, 5): {
+        'chrome99': ImpersonateTarget('chrome', '99', 'windows', '10'),
+        'chrome99_android': ImpersonateTarget('chrome', '99', 'android', '12'),
+        'chrome100': ImpersonateTarget('chrome', '100', 'windows', '10'),
+        'chrome101': ImpersonateTarget('chrome', '101', 'windows', '10'),
+        'chrome104': ImpersonateTarget('chrome', '104', 'windows', '10'),
+        'chrome107': ImpersonateTarget('chrome', '107', 'windows', '10'),
+        'chrome110': ImpersonateTarget('chrome', '110', 'windows', '10'),
+        'edge99': ImpersonateTarget('edge', '99', 'windows', '10'),
+        'edge101': ImpersonateTarget('edge', '101', 'windows', '10'),
+        'safari15_3': ImpersonateTarget('safari', '15.3', 'macos', '11'),
+        'safari15_5': ImpersonateTarget('safari', '15.5', 'macos', '12'),
```

Comment:
This change results in `chrome-99:android-12` being selected by default when `impersonate=True` or `impersonate='chrome'` is used in an extractor request. The previous default selection was `chrome-110:windows-10`. IMO we shouldn't change the default

Reply:
we should also consider the defaults selected in 0.10.x